recipe/patches/0002-fix-lemon.patch

```diff
@@ -0,0 +1,93 @@
+From e219714ad2f64045ee9aeb7d3ed3926de77657c1 Mon Sep 17 00:00:00 2001
+From: Lorenzo Pirritano <lpirritano@anaconda.com>
+Date: Thu, 20 Jun 2024 21:27:04 +0200
+Subject: [PATCH 2/2] fix lemon
+
+---
+ cmake/FindLEMON.cmake        | 13 +++++++------
+ src/cc/CMakeLists.txt        |  2 +-
+ src/cc/dotnet/CMakeLists.txt |  2 +-
+ src/cc/python/CMakeLists.txt |  2 +-
+ 4 files changed, 10 insertions(+), 9 deletions(-)
+
+diff --git a/cmake/FindLEMON.cmake b/cmake/FindLEMON.cmake
+index a3045d2..4a38afb 100644
+--- a/cmake/FindLEMON.cmake
++++ b/cmake/FindLEMON.cmake
```

Comment:
This feels weird. The LEMON package already has a CMake config file. If cmake is not able to find it, it's either because the path is not in `CMAKE_MODULE_PATH`/`CMAKE_PREFIX_PATH` or we need to set https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_PREFER_CONFIG.html#variable:CMAKE_FIND_PACKAGE_PREFER_CONFIG.

Reply:
Ah I see why they do this (I think). Lemon's cmake config file doesn't export any target and only exports `LEMON_LIBRARY` and `LEMON_LIBRARIES` variables.

Reply:
Anyway, we should patch `cmake/FindLEMON.cmake` to export `LEMON::LEMON` instead of `lemon::lemon`, which will remove the need to patch `src/cc/CMakeLists.txt` and `src/cc/python/CMakeLists.txt`.

Reply:
Will do, thanks for the advice